woocommerce/woocommerce-android
# [TTP] Handle TTP unavailable when universal link is clicked

Closes: #9347


### Description

The PR adds better handling of the situation when device is not supported for TTP. We offer to purchase a reader in most case

### Testing instructions

* Modify code `TapToPayAvailabilityStatus`

```
    operator fun invoke() =
        if (true) Result.NotAvailable.SystemVersionNotSupported (put all 5 statuses here to test) else 
        when {
            !appPrefs.isTapToPayEnabled -> Result.NotAvailable.TapToPayDisabled
            !systemVersionUtilsWrapper.isAtLeastQ() -> Result.NotAvailable.SystemVersionNotSupported
            !deviceFeatures.isGooglePlayServicesAvailable() -> Result.NotAvailable.GooglePlayServicesNotAvailable
            !deviceFeatures.isNFCAvailable() -> Result.NotAvailable.NfcNotAvailable
            !isTppSupportedInCountry(wooStore.getStoreCountryCode(selectedSite.get())) ->
                Result.NotAvailable.CountryNotSupported

            else -> Result.Available
        }
```
* Try all five unavailable statuses by  running To simulate a universal link click
```
adb shell am start -a android.intent.action.VIEW -c android.intent.category.BROWSABLE -d "https://woocommerce.com/mobile/payments/tap-to-pay"
```
* Notice new dialogs

### Images/gif

https://github.com/woocommerce/woocommerce-android/assets/4923871/ae8bc2d2-2303-4b13-bbfe-fe9ad3612f45
